orangecontrib/text/import_documents.py

```diff
@@ -345,14 +345,9 @@ def _add_metadata(self, corpus: Corpus) -> Corpus:
         if len(df.index.drop_duplicates()) != len(df.index):
             df = df[~df.index.duplicated(keep='first')]
         filtered = df.reindex(path_column)
-        for column in filtered.columns:
-            corpus = corpus.add_column(
-                StringVariable(get_unique_names(corpus.domain, column)),
-                filtered[column].to_numpy(),
-                to_metas=True
-            )
+        filtered = table_from_frame(filtered)
```

Comment:
If you want to use the `table_from_frame`, you need to reset the df.index:

`filtered = table_from_frame(filtered.reset_index(drop=True))`

Reply:
I'll try the above suggestion first, thanks! ^^